packages/devtools/lib/src/debugger/console_area.dart

```diff
@@ -54,19 +59,83 @@ class ConsoleArea implements CoreElementView {
     });
   }
 
+  Position _getDocumentEnd() {
+    final doc = _editor.getDoc();
+    final lastLine = doc.lastLine();
+    return Position(lastLine, doc.getLine(lastLine).length);
+  }
+
   void _append(String text) {
-    // append text
-    _editor
-        .getDoc()
-        .replaceRange(text, Position(_editor.getDoc().lastLine() + 1, 0));
-
-    // scroll to end
-    final int lastLineIndex = _editor.getDoc().lastLine();
-    final String lastLine = _editor.getDoc().getLine(lastLineIndex);
-    _editor.scrollIntoView(lastLineIndex, lastLine.length);
+    final scrollInfo = _editor.getScrollInfo();
+    // If the last line of the console is visible, after appending content we
+    // should scroll so that the last line of the editor is still visible.
+    // TODO(jacobr): add an optional setting to always scroll to the end of the
+    // console when new content is added.
+    final scrollToEnd =
+        scrollInfo.top + scrollInfo.clientHeight >= scrollInfo.height;
+
+    final chunks = decodeAnsiColorEscapeCodes(text, _ansiUp);
+
+    final doc = _editor.getDoc();
+    for (StyledText chunk in chunks) {
```

Comment:
[optional] Consider `var` over `StyledText`

Reply:
think the type is valuable here as it isn't obvious from the code what the type is and it didn't add many chars.